indra/llcommon/llerror.cpp

```diff
@@ -1044,7 +1045,7 @@ namespace LLError
             return;
         }
         SettingsConfigPtr s = Globals::getInstance()->getSettingsConfig();
-        LLCoros::LockType lock(s->mRecorderMutex);
+        std::unique_lock<boost::fibers::recursive_mutex> lock(s->mRecorderMutex);
```

Comment:
C++17 has Class Template Argument Deduction. Let's try declaring just `std::unique_lock` here. If that works, I'd change `LLCoros::LockType` to just `std::unique_lock`.